### PR TITLE
CA-150197: Improvement: No confirmation message before delete wlb scheduler

### PR DESCRIPTION
**Pull request by fillzero**
Add confirm dialog before delete WLB Optimization Mode Scheduler.

add_confirm_dialog_before_delete

Signed-off-by: Liang Dai <liang.dai1@citrix.com>

### CODE REVIEW

File: XenAdmin/Controls/Wlb/WlbOptModeScheduler.cs [hunk 1]
```diff
@@ -646,8 +646,17 @@ private void buttonDelete_Click(object sender, EventArgs e)
         {
             if (lvTaskList.SelectedItems.Count > 0)
             {
-                WlbScheduledTask task = TaskFromItem(lvTaskList.SelectedItems[0]);
-                DeleteTask(task);
+                var confirmResult = MessageBox.Show(this,
```
Comment (MihaelaStoica) [inline XenAdmin/Controls/Wlb/WlbOptModeScheduler.cs]: We usually use a ThreeButtonDialog for confirmation messages, for example:
```
var confirmResult = new ThreeButtonDialog(
                                new ThreeButtonDialog.Details(SystemIcons.Warning, 
                                                               Messages.DELETE_WLB_OPTIMIZATION_SCHEDULE_WARNING, 
                                                               Messages.DELETE_WLB_OPTIMIZATION_SCHEDULE_CAPTION),
                                ThreeButtonDialog.ButtonYes,
                                ThreeButtonDialog.ButtonNo).ShowDialog(this);
```

File: XenAdmin/Controls/Wlb/WlbOptModeScheduler.cs [hunk 2]
```diff
@@ -646,8 +646,17 @@ private void buttonDelete_Click(object sender, EventArgs e)
...
+                                                    Messages.DELETE_WLB_OPTIMIZATION_SCHEDULE_WARNING,
+                                                    Messages.DELETE_WLB_OPTIMIZATION_SCHEDULE_CAPTION,
+                                                    MessageBoxButtons.YesNo,
+                                                    MessageBoxIcon.Warning, MessageBoxDefaultButton.Button1);
+
+                if (confirmResult == DialogResult.Yes)
+                {
+                    WlbScheduledTask task = TaskFromItem(lvTaskList.SelectedItems[0]);
+                    DeleteTask(task);
+                }
```
Comment (MihaelaStoica) [inline XenAdmin/Controls/Wlb/WlbOptModeScheduler.cs]: I know this not part of your change, but I believe that the `weekView1.Refresh()` call should be inside the `if` block, as there is no need to refresh if nothing has chanhed.

**Comment by fillzero** [comment]
Hi @MihaelaStoica 
Thanks for you review and the comments
I have updated a new commit.

**Comment by MihaelaStoica** [comment]
Looks good. Merging.

**Pull request merged**